server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreClusterStateRestoreIT.java

```diff
@@ -0,0 +1,280 @@
+        resetCluster(dataNodeCount, clusterManagerNodeCount);
+
+        String newClusterUUID = clusterService().state().metadata().clusterUUID();
+        assert !Objects.equals(newClusterUUID, prevClusterUUID) : "cluster restart not successful. cluster uuid is same";
+
+        // Step - 3 Trigger full cluster restore and validate
+        restoreAndValidate(prevClusterUUID, indexStats);
+    }
+
+    @AwaitsFix(bugUrl = "waiting upload flow rebase. tested on integration PR")
+    public void testFullClusterRestoreMultipleIndices() throws Exception {
+        int shardCount = randomIntBetween(1, 2);
+        int replicaCount = 1;
+        int dataNodeCount = shardCount * (replicaCount + 1);
+        int clusterManagerNodeCount = 1;
+
+        // Step - 1 index some data to generate files in remote directory
+        Map<String, Long> indexStats = initialTestSetup(shardCount, replicaCount, dataNodeCount, clusterManagerNodeCount);
+
+        String secondIndexName = INDEX_NAME + "-2";
+        createIndex(secondIndexName, remoteStoreIndexSettings(replicaCount, shardCount + 1));
+        Map<String, Long> indexStats2 = indexData(1, false, secondIndexName);
+        assertEquals((shardCount + 1) * (replicaCount + 1), getNumShards(secondIndexName).totalNumShards);
+        ensureGreen(secondIndexName);
+
+        String prevClusterUUID = clusterService().state().metadata().clusterUUID();
+
+        // Step - 2 Replace all nodes in the cluster with new nodes. This ensures new cluster state doesn't have previous index metadata
+        resetCluster(dataNodeCount, clusterManagerNodeCount);
+
+        String newClusterUUID = clusterService().state().metadata().clusterUUID();
+        assert !Objects.equals(newClusterUUID, prevClusterUUID) : "cluster restart not successful. cluster uuid is same";
+
+        // Step - 3 Trigger full cluster restore
+        restoreAndValidate(prevClusterUUID, indexStats);
+        ensureGreen(secondIndexName);
+        verifyRestoredData(indexStats2, secondIndexName);
+    }
+
+    @AwaitsFix(bugUrl = "waiting upload flow rebase. tested on integration PR")
+    public void testFullClusterRestoreFailureValidationFailures() throws Exception {
+        int shardCount = randomIntBetween(1, 2);
+        int replicaCount = 1;
+        int dataNodeCount = shardCount * (replicaCount + 1);
+        int clusterManagerNodeCount = 1;
+
+        // index some data to generate files in remote directory
+        Map<String, Long> indexStats = initialTestSetup(shardCount, replicaCount, dataNodeCount, clusterManagerNodeCount);
+        String prevClusterUUID = clusterService().state().metadata().clusterUUID();
+
+        // Start of Test - 1
+        // Test - 1 Trigger full cluster restore and validate it fails due to incorrect cluster UUID
+        PlainActionFuture<RestoreRemoteStoreResponse> future = PlainActionFuture.newFuture();
+        restoreAndValidateFails("randomUUID", future);
```

Comment:
you can pass the exception you expect for a particular failure

Reply:
sure, done